diff --git a/node_modules/next/dist/build/webpack/plugins/next-types-plugin.js b/node_modules/next/dist/build/webpack/plugins/next-types-plugin.js
index 8532ab0..7b8914b 100644
--- a/node_modules/next/dist/build/webpack/plugins/next-types-plugin.js
+++ b/node_modules/next/dist/build/webpack/plugins/next-types-plugin.js
@@ -273,7 +273,7 @@ class NextTypesPlugin {
         if (!this.typedRoutes) return;
         const isApp = filePath.startsWith(this.appDir + _path.default.sep);
         // Filter out non-page files in app dir
-        if (isApp && !/[/\\]page\.[^.]+$/.test(filePath)) {
+        if (isApp && !/[/\\](?:page|route)\.[^.]+$/.test(filePath)) {
             return;
         }
         // Filter out non-page files in pages dir
diff --git a/node_modules/next/dist/client/components/layout-router.js b/node_modules/next/dist/client/components/layout-router.js
index 9b60a45..dd0639d 100644
--- a/node_modules/next/dist/client/components/layout-router.js
+++ b/node_modules/next/dist/client/components/layout-router.js
@@ -317,6 +317,7 @@ function HandleRedirect({ redirect  }) {
     const router = (0, _navigation).useRouter();
     (0, _react).useEffect(()=>{
         router.replace(redirect, {});
+        router.refresh()
     }, [
         redirect,
         router
diff --git a/node_modules/next/dist/client/link.js b/node_modules/next/dist/client/link.js
index d15ce7f..369e036 100644
--- a/node_modules/next/dist/client/link.js
+++ b/node_modules/next/dist/client/link.js
@@ -83,6 +83,7 @@ function linkClicked(e, router, href, as, replace, shallow, scroll, locale, isAp
     if (isAppRouter) {
         // @ts-expect-error startTransition exists.
         _react.default.startTransition(navigate);
+        router.refresh()
     } else {
         navigate();
     }
